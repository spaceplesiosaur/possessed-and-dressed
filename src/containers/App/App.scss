
* {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
}

.app {
  text-align: center;
}
