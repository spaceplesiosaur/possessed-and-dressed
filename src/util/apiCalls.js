export const getInfo = (url, type) => {
  const options = {
    method: 'GET',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': 'Token 3fc50009100f8a4794408699493b41c3b84467f5',
    }
  }
  return fetch(url, options)
    .then(response => {
      if(!response.ok) {
        throw Error (`Oh no! There is a problem finding ${type}`)
      }
      return response.json()
    })
}
