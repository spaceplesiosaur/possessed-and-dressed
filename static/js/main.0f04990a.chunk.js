(this["webpackJsonppossessed-and-dressed"]=this["webpackJsonppossessed-and-dressed"]||[]).push([[0],{31:function(e,t,n){e.exports=n(52)},36:function(e,t,n){},37:function(e,t,n){},38:function(e,t,n){},44:function(e,t,n){},45:function(e,t,n){},47:function(e,t,n){},48:function(e,t,n){},49:function(e,t,n){},50:function(e,t,n){},51:function(e,t,n){},52:function(e,t,n){"use strict";n.r(t);var o=n(0),a=n.n(o),s=n(19),r=n.n(s),c=(n(36),n(7)),i=n(8),l=n(10),u=n(9),h=n(11),p=(n(37),n(38),n(14)),m=function(){return a.a.createElement(p.b,{to:"/"},a.a.createElement("header",{className:"app-header"},"Posessed & Dressed"))},d=n(12),f=n.n(d),C=n(24),E=n(16),b=(n(44),n(45),function(e){function t(e){var n;e.picture,e.name,e.happy_pic,e.change,e.click;return Object(c.a)(this,t),(n=Object(l.a)(this,Object(u.a)(t).call(this))).switchPic=function(){return!0===n.state.ishappy?n.props.happy_pic:!1===n.state.ishappy?n.props.picture:void 0},n.scareHost=function(){n.setState({ishappy:!1})},n.unScareHost=function(){n.setState({ishappy:!0})},n.clickChooseHost=function(e){return f.a.async((function(t){for(;;)switch(t.prev=t.next){case 0:return t.next=2,f.a.awrap(n.props.change(e));case 2:return t.next=4,f.a.awrap(n.props.click());case 4:case"end":return t.stop()}}))},n.state={ishappy:!0},n}return Object(h.a)(t,e),Object(i.a)(t,[{key:"render",value:function(){return a.a.createElement("div",{className:"entryPage-miniHost-hostAvatar"},a.a.createElement("img",{src:this.switchPic(),className:"host-avatar-picture",alt:"host-screaming",onMouseEnter:this.scareHost,onMouseLeave:this.unScareHost,name:"hostName","data-host":this.props.name,onClick:this.clickChooseHost}),a.a.createElement("p",{className:"host-avatar-name"},this.props.name))}}]),t}(o.Component)),y=n(4),g=function(e){function t(){var e;return Object(c.a)(this,t),(e=Object(l.a)(this,Object(u.a)(t).call(this))).generateHostPics=function(){return e.props.hostList.map((function(t){return a.a.createElement(b,{key:t.id,picture:t.picture,happy_pic:t.happy_picture,name:t.name,change:e.handleImageClick,click:e.handleClick,id:t.id})}))},e.handleChange=function(t){e.setState(Object(C.a)({},t.target.name,t.target.value))},e.handleImageClick=function(t){e.setState(Object(C.a)({},t.target.name,t.target.dataset.host))},e.findHost=function(){var t=e.props.hostList.find((function(t){return t.name===e.state.hostName}));return t||"error"},e.handleError=function(){e.state.error&&e.setState({inputClass:{input:"entryForm-hostName-inputError",text:".entryForm-error-text"}})},e.handleClick=function(){return f.a.async((function(t){for(;;)switch(t.prev=t.next){case 0:if("error"!==e.findHost()&&""!==e.state.hostName){t.next=6;break}return t.next=3,f.a.awrap(e.setState({error:!0}));case 3:e.handleError(),t.next=8;break;case 6:e.props.chooseAHost(e.findHost()),e.setState({hostPossessed:!0});case 8:case"end":return t.stop()}}))},e.state={hostName:"",error:!1,inputClass:{input:"entryForm-hostName-input",text:"hidden"},hostPossessed:!1},e}return Object(h.a)(t,e),Object(i.a)(t,[{key:"render",value:function(){return this.props.host.name?a.a.createElement(E.a,{to:"/hosts/".concat(this.props.host.id)}):a.a.createElement("section",{className:"app-entryForm"},a.a.createElement("h2",null,"Welcome!  Choose the person you would like to possess"),a.a.createElement("div",{className:"entryForm-hostName-inputSet"},a.a.createElement("div",{className:"entryForm-hostName-styling"},a.a.createElement("label",{htmlFor:"hostName",className:"entryForm-hostName-label"},"Chosen Host:"),a.a.createElement("input",{name:"hostName",id:"hostName-input",className:this.state.inputClass.input,onChange:this.handleChange}),a.a.createElement("button",{className:"choose-host-button",onClick:this.handleClick,id:"hostName-button"},"Dress & Possess!")),a.a.createElement("p",{className:this.state.inputClass.text},"Only enter a name from the list!")),a.a.createElement("section",{className:"app-entryForm-hostList"},this.generateHostPics()))}}]),t}(o.Component),v=Object(y.b)((function(e){return{hostList:e.hosts,host:e.chosenHost}}),(function(e){return{chooseAHost:function(t){return e(function(e){return{type:"CHOOSE_HOST",chosenHost:e}}(t))}}}))(g),O=(n(47),n(48),function(e){function t(e){var n;e.colorName,e.hues,e.borderColor;return Object(c.a)(this,t),(n=Object(l.a)(this,Object(u.a)(t).call(this))).toggleListDisplay=function(){n.setState({open:!n.state.open})},n.handleMenuClick=function(){return f.a.async((function(e){for(;;)switch(e.prev=e.next){case 0:return e.next=2,f.a.awrap(n.toggleListDisplay());case 2:n.state.open?n.setState({listDisplay:"colorList-hues"}):n.setState({listDisplay:"hidden"});case 3:case"end":return e.stop()}}))},n.handleColorClick=function(e){var t=n.props.hues.find((function(t){return t.name===e.target.id}));n.props.chooseColor(t)},n.hydrateColorList=function(){return n.props.hues.map((function(e){return a.a.createElement("li",{className:"colorList-box",style:{backgroundColor:"".concat(e.hex_code)},id:e.name,onClick:n.handleColorClick,key:e.id})}))},n.state={open:!1,listDisplay:"hidden"},n}return Object(h.a)(t,e),Object(i.a)(t,[{key:"render",value:function(){return""!==this.props.chosenColor?a.a.createElement(E.a,{to:"/hosts/".concat(this.props.chosenHost.id,"/").concat(this.props.chosenColor.name)}):a.a.createElement("div",{className:"colorChooser-colorDropdown-widget"},a.a.createElement("h4",{className:"colorList-box",style:{border:"10px solid ".concat(this.props.borderColor)},onClick:this.handleMenuClick,id:"menu-header"},this.props.colorName),a.a.createElement("ul",{className:this.state.listDisplay},this.hydrateColorList()))}}]),t}(o.Component)),k=Object(y.b)((function(e){return{chosenHost:e.chosenHost,chosenColor:e.chosenColor}}),(function(e){return{chooseColor:function(t){return e({type:"CHOOSE_COLOR",chosenColor:t})}}}))(O),N=function(e,t){return fetch(e,{method:"GET",headers:{"Content-Type":"application/json",Authorization:"Token 3fc50009100f8a4794408699493b41c3b84467f5"}}).then((function(e){if(!e.ok)throw Error("Oh no! There is a problem finding ".concat(t));return e.json()}))},S=function(e){function t(e){var n;e.colorCategories,e.hexCodes;return Object(c.a)(this,t),(n=Object(l.a)(this,Object(u.a)(t).call(this))).supplyColorCategories=function(){N("https://color-seasons.herokuapp.com/categories/","color categories").then((function(e){return n.props.storeCategories(e)}))},n.supplyHexCodes=function(){N("https://color-seasons.herokuapp.com/colors/","colors and hex codes").then((function(e){return n.props.storeAllColors(e)}))},n.findMenuParentColor=function(e,t){return e.find((function(e){return Object.keys(e)[0]===t}))[t]},n.matchAvailableHuesToList=function(e){return e.map((function(e){return n.props.hexCodes.find((function(t){return t.id===e}))}))},n.displayColorBar=function(){var e=[{red:"#F87E97"},{orange:"#F89C51"},{yellow:"#E7C142"},{green:"#A7C65F"},{blue:"#5DA1DA"},{violet:"#5C50B6"},{pink:"#DE92DA"},{neutral:"#E2E2E3"}];return n.props.colorCategories.map((function(t){var o=n.findMenuParentColor(e,t.name),s=n.matchAvailableHuesToList(t.colors);return s.every((function(e){return void 0!==e}))&&a.a.createElement(k,{key:t.id,colorName:t.name,hues:s,borderColor:o})}))},n.state={},n}return Object(h.a)(t,e),Object(i.a)(t,[{key:"componentDidMount",value:function(){this.supplyColorCategories(),this.supplyHexCodes()}},{key:"render",value:function(){return a.a.createElement("section",{className:"colorBar-main-frame"},a.a.createElement("h2",{className:"colorBar-main-header"},"What color would you like your host to wear?"),a.a.createElement("p",{className:"colorBar-main-explantion"},"Remember, it is VERY IMPORTANT to dress your host in something they look their best in - if they don't look their best, the church will become suspicious and may perform an exorcism on them!"),a.a.createElement("section",{className:"colorBar-frame-navBar"},this.displayColorBar()))}}]),t}(o.Component),H=Object(y.b)((function(e){return{colorCategories:e.categories,hexCodes:e.allColors,seasons:e.seasons}}),(function(e){return{storeCategories:function(t){return e(function(e){return{type:"SET_CATEGORIES",categories:e}}(t))},storeAllColors:function(t){return e({type:"SET_ALL_COLORS",colors:t})}}}))(S),j=(n(49),function(e){var t=e.host;return a.a.createElement("section",{className:"hostPage-main-frame"},a.a.createElement("p",{className:"host-name-tag"},"Meet ",t.name,"! ",t.name," is very excited to be possessed by you."),a.a.createElement("img",{className:"hostPage-picture-host",src:t.picture,alt:"hostPage-main-hostPic"}))}),w=(n(50),Object(y.b)((function(e){return{chosenColor:e.chosenColor,chosenHost:e.chosenHost}}),null)((function(e){var t=e.match,n=e.chosenColor,o=(e.chosenHost,e.season);return n.name&&a.a.createElement("section",{className:"colorFeedback-main"},a.a.createElement("div",{className:"colorFeedback-main-colorBox",style:{backgroundColor:"".concat(n.hex_code)}}),a.a.createElement("h2",{className:"colorFeedback-main-name"},n.name.toUpperCase()),t?a.a.createElement("p",{className:"colorFeedback-main-feedback"},"Great choice! Your host's season is ","".concat(o.toUpperCase()),". ","".concat(n.name.toUpperCase())," will look great on them - nobody will suspect a thing!  Happy haunting!"):a.a.createElement("p",{className:"colorFeedback-main-feedback"},"Terrible choice! Your host's season is ","".concat(o.toUpperCase()),". ","".concat(n.name.toUpperCase())," will make them look tired, sad and afraid, and the church will surely try to perform an exorcism on them!"),a.a.createElement("button",{className:"back-button",onClick:function(){return window.location="/"}},"Back"))}))),x=(n(51),function(){return a.a.createElement("p",{className:"error-page"},"404 - This page is not found!!")}),L=function(e){function t(){var e;return Object(c.a)(this,t),(e=Object(l.a)(this,Object(u.a)(t).call(this))).supplyHosts=function(){return N("https://color-seasons.herokuapp.com/hosts/","hosts").then((function(t){return e.props.storeHosts(t)}))},e.supplySeasons=function(){N("https://color-seasons.herokuapp.com/seasons/","seasonal analysis").then((function(t){return e.props.storeSeasons(t)}))},e.determineHostSeason=function(){return e.props.seasonList.find((function(t){return t.id===e.props.chosenHost.season}))},e.determineMatch=function(){return e.determineHostSeason().colors.includes(e.props.chosenColor.id)},e.state={},e}return Object(h.a)(t,e),Object(i.a)(t,[{key:"componentDidMount",value:function(){this.supplyHosts(),this.supplySeasons()}},{key:"render",value:function(){var e=this;return a.a.createElement("main",{className:"app"},a.a.createElement(E.d,null,a.a.createElement(E.b,{exact:!0,path:"/",render:function(){return a.a.createElement(a.a.Fragment,null,a.a.createElement(m,null),a.a.createElement(v,null))}}),a.a.createElement(E.b,{exact:!0,path:"/hosts/:id",render:function(t){var n=e.props.hostList.find((function(e){return e.id===parseInt(t.match.params.id)}));return n&&a.a.createElement(a.a.Fragment,null,a.a.createElement(m,null),a.a.createElement(j,{host:n}),a.a.createElement(H,null))}}),a.a.createElement(E.b,{exact:!0,path:"/hosts/:id/:colorName",render:function(t){return a.a.createElement(a.a.Fragment,null,a.a.createElement(m,null),a.a.createElement(w,{match:e.determineMatch(),season:e.determineHostSeason().name}))}}),a.a.createElement(E.b,{path:"*",render:function(){return a.a.createElement(a.a.Fragment,null,a.a.createElement(m,null),a.a.createElement(x,null))}})))}}]),t}(o.Component),T=Object(y.b)((function(e){return{hostList:e.hosts,seasonList:e.seasons,chosenHost:e.chosenHost,chosenColor:e.chosenColor,hexList:e.allColors}}),(function(e){return{storeHosts:function(t){return e({type:"SET_HOSTS",hosts:t})},storeSeasons:function(t){return e({type:"SET_SEASONS",seasons:t})}}}))(L),F=n(17),_=n(30),A=Object(F.combineReducers)({hosts:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_HOSTS":return t.hosts;default:return e}},chosenHost:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:{},t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"CHOOSE_HOST":return t.chosenHost;default:return e}},categories:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_CATEGORIES":return t.categories;default:return e}},seasons:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_SEASONS":return t.seasons;default:return e}},allColors:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:[],t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"SET_ALL_COLORS":return t.colors;default:return e}},chosenColor:function(){var e=arguments.length>0&&void 0!==arguments[0]?arguments[0]:"",t=arguments.length>1?arguments[1]:void 0;switch(t.type){case"CHOOSE_COLOR":return t.chosenColor;default:return e}}});Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));var D=Object(F.createStore)(A,Object(_.composeWithDevTools)());r.a.render(a.a.createElement(y.a,{store:D},a.a.createElement(p.a,{basename:"/possessed-and-dressed"},a.a.createElement(T,null))),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then((function(e){e.unregister()}))}},[[31,1,2]]]);
//# sourceMappingURL=main.0f04990a.chunk.js.map