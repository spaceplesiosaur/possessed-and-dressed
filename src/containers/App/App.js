import React, { Component } from 'react';
import './App.scss';
import Header from '../../components/Header/Header';
import EntryForm from '../EntryForm/EntryForm';
import HostPage from '../../components/HostPage/HostPage';
import { Route } from 'react-router-dom';
import { connect } from 'react-redux';
import { getInfo } from '../../util/apiCalls';
import { setHosts } from '../../actions/index.js';

export class App extends Component {
  constructor() {
    super()
    this.state = {

    }
  }

  componentDidMount() {
    this.supplyHosts()
  }

  supplyHosts = () => {
    return getInfo('https://color-seasons.herokuapp.com/hosts/', 'hosts')
    .then(data => this.props.storeHosts(data))
  }

  render() {
    return (
      <main className="app">
        <Route exact path='/' render={() => {
          return (
            <>
              <Header />
              <EntryForm />
            </>
          )
        }} />
        <Route path='/hosts/:id' render={(match) => {
          console.log('MATCH', match.match.params)
          const selectedHost = this.props.hostList.find(host => {
            console.log('ID', host.id)
            return host.id === parseInt(match.match.params.id)
          })

          return (
            selectedHost && <HostPage host={selectedHost}/>
          )
        }}/>
      </main>
    );
  }
}

export const mapStateToProps = (state) => ({
  hostList: state.hosts,
  chosenHost: state.chosenHost
})

export const mapDispatchToProps = (dispatch) => ({
  storeHosts: hostData => dispatch(setHosts(hostData))
})
export default connect(mapStateToProps, mapDispatchToProps)(App)
