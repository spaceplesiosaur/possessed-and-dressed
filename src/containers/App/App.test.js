import React from 'react';
import { render } from '@testing-library/react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { getInfo } from '../../util/apiCalls.js';
import { setHosts } from '../../actions/index.js';
import { App } from './App';

jest.mock('../../util/apiCalls.js');

describe('App', () => {
  let wrapper;
  let mockHosts;

  beforeEach(() => {
    getInfo.mockImplementation(() => {
      return Promise.resolve(mockHosts)
    })
    
    mockHosts = [
    {
        "id": 1,
        "name": "Blucifer",
        "picture": "https://i.imgur.com/I5MqMUV.jpg",
        "season": 9
    },
    {
        "id": 2,
        "name": "Shimmeree",
        "picture": "https://i.imgur.com/P9srxtU.jpg",
        "season": 6
    },
    {
        "id": 3,
        "name": "Rainbow Dash",
        "picture": "https://i.imgur.com/GJthKTQ.jpg",
        "season": 7
    }
    ]

    wrapper = shallow(<App
        hostList={mockHosts}
        storeHosts={jest.fn()}
      />)
  })

  it('should render App with correct data', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should call supplyHosts when component mounts', () => {
    wrapper.instance().supplyHosts = jest.fn()
    wrapper.instance().componentDidMount()

    expect(wrapper.instance().supplyHosts).toHaveBeenCalled()
  })

  it('should call getInfo and storeHosts when supplyHosts is called', async () => {

    await wrapper.instance().supplyHosts()

    expect(wrapper.instance().props.storeHosts).toHaveBeenCalled()
    expect(getInfo).toHaveBeenCalled()
  })
})
