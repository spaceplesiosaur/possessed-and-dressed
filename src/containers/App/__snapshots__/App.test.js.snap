// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App with correct data 1`] = `
<main
  className="app"
>
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
</main>
`;
